src/Compiler/Checking/import.fs

```diff
@@ -63,12 +63,17 @@ type TTypeCacheKey =
     val ty1: TType
     val ty2: TType
     val canCoerce: CanCoerce
+    val hashCode: int
 
-    private new (ty1, ty2, canCoerce) =
-        { ty1 = ty1; ty2 = ty2; canCoerce = canCoerce }
+    private new (ty1, ty2, canCoerce, hashCode) =
+        { ty1 = ty1; ty2 = ty2; canCoerce = canCoerce; hashCode = hashCode }
 
     static member FromStrippedTypes (ty1, ty2, canCoerce) =
-        TTypeCacheKey(ty1, ty2, canCoerce)
+        let hashCode =
```

Comment:
Since this is supposed to be immutable and the key is immediately consumed, it makes sense to precompute the hash.
OTOH it may be a premature optimization and can make some problems more obscure.

Reply:
I would have assumed that the Dictionary storage calls into that once only anyway and then stores it, but it likely does not matter.

Reply:
https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L485

It's called each time. That's why I ran into problems initially, when the hashcode depended on some mutable parts of TType.